plugins/woocommerce/src/Internal/Admin/ProductReviews/ReviewsUtil.php

```diff
@@ -10,17 +10,52 @@ class ReviewsUtil {
 	/**
 	 * Removes product reviews from the edit-comments page to fix the "Mine" tab counter.
 	 *
-	 * @param  array|mixed $clauses A compacted array of comment query clauses.
+	 * @param array|mixed       $clauses A compacted array of comment query clauses.
+	 * @param \WP_Comment_Query $comment_query The WP_Comment_Query instance being filtered.
+	 *
 	 * @return array|mixed
 	 */
-	public static function comments_clauses_without_product_reviews( $clauses ) {
-		global $wpdb, $current_screen;
+	public static function comments_clauses_without_product_reviews( $clauses, $comment_query ) {
+		global $wpdb;
+
+		if ( ! empty( $comment_query->query_vars['post_type'] ) ) {
+			$post_type = $comment_query->query_vars['post_type'];
+			if ( ! is_array( $post_type ) ) {
+				$post_type = explode( ',', $post_type );
+			}
+			if ( ! in_array( 'product', $post_type ) ) {
+				return $clauses;
+			}
+		}
+
+		/**
+		 * Any comment queries with these values are likely to be custom handling where we don't want to change default behavior.
+		 */
+		foreach ( [ 'ID', 'parent', 'parent__in', 'post_author__in', 'post_author', 'post_name', 'type', 'type__in', 'type__not_in', 'post_type__in' ] as $arg ) {
+			if ( ! empty( $comment_query->query_vars[ $arg ] ) ) {
+				return $clauses;
+			}
+		}
+
+		if ( ! empty( $comment_query->query_vars['post_id'] ) && absint( $comment_query->query_vars['post_id'] ) > 0 ) {
+			if ( 'product' === get_post_type( absint( $comment_query->query_vars['post_id'] ) ) ) {
+				return $clauses;
+			}
+		}
 
-		if ( isset( $current_screen->base ) && 'edit-comments' === $current_screen->base ) {
-			$clauses['join']  .= " LEFT JOIN {$wpdb->posts} AS wp_posts_to_exclude_reviews ON comment_post_ID = wp_posts_to_exclude_reviews.ID ";
-			$clauses['where'] .= ( $clauses['where'] ? ' AND ' : '' ) . " wp_posts_to_exclude_reviews.post_type NOT IN ('product') ";
+		if ( ! empty( $comment_query->query_vars['post__in'] ) ) {
+			$post_ids = wp_parse_id_list( $comment_query->query_vars['post__in'] );
+			_prime_post_caches( $post_ids );
+			foreach ( $post_ids as $post_id ) {
+				if ( 'product' === get_post_type( $post_id ) ) {
+					return $clauses;
+				}
+			}
 		}
 
+		$clauses['join']  .= " LEFT JOIN {$wpdb->posts} AS wp_posts_to_exclude_reviews ON comment_post_ID = wp_posts_to_exclude_reviews.ID ";
+		$clauses['where'] .= ( $clauses['where'] ? ' AND ' : '' ) . " wp_posts_to_exclude_reviews.post_type NOT IN ('product') ";
```

Comment:
I'm a little surprised we don't wrap the `$clauses['where']` with a `trim()` for safety, but this exact same logic is used in a bunch of other places across this repo and others without issue, so it may not be a likely scenario.

Reply:
Good call. I'm adding trim() to all the usages since comments are covered in this PR to prevent any future/current cases where another plugin adding a value like `' '` causing a failed query.